src/simdutf/rvv/implementation.h

```diff
@@ -0,0 +1,110 @@
+#ifndef SIMDUTF_RVV_IMPLEMENTATION_H
+#define SIMDUTF_RVV_IMPLEMENTATION_H
+
+#include "simdutf.h"
+#include "simdutf/internal/isadetection.h"
+
+namespace simdutf {
+namespace rvv {
+
+namespace {
+using namespace simdutf;
+} // namespace
+
+class implementation final : public simdutf::implementation {
+public:
+  simdutf_really_inline implementation()
+      : simdutf::implementation("rvv", "RISC-V Vector Extension",
+                                 internal::instruction_set::RVV)
+      , _supports_zvbb(
+#if SIMDUTF_IS_ZVBB
```

Comment:
We should relay on `detect_supported_architectures` result. I think we can remove this `ifdef`.

Reply:
Yes, but I tried to avoid vcompress when possible, because it's performance varies widely on current platforms.
I'd leave the code as is, but this is certainly something to reconsider once there is more hardware available.

Approximation from my measurements (vncvt is vnsrl.vi/vnsra.vi):
* [C908](https://camel-cdr.github.io/rvv-bench-results/canmv_k230/index.html): vncvt_m4+vncvt_m2: 12 cycles, vcompress_m4 36 cycles.
* [C920](https://camel-cdr.github.io/rvv-bench-results/milkv_pioneer/index.html): vncvt_m4+vncvt_m2: 6 cycles, vcompress_m4 5.4 cycles.

Reply:
@camel-cdr 

> if SIMDUTF_IS_ZVBB is true, then we actually just want to return true, because the code won't run on a platform where zvbb isn't available if it's enabled globally.

Don't you then want something like...


```C++
#if SIMDUTF_IS_ZVBB

static inline uint32_t detect_supported_architectures() {
  return internal::instruction_set::ZVBB;
}

#endif
```

like...

https://github.com/simdutf/simdutf/blob/73ac74e266fa6e6c07fa2693e1effd9cdc39bee9/include/simdutf/internal/isadetection.h#L87

Reply:
I see that you already have...

```C++
 #if SIMDUTF_IS_ZVBB
   host_isa |= instruction_set::ZVBB;
 #endif
```

So unless I am missing something, you already have that `internal::detect_supported_architectures() & internal::instruction_set::ZVBB` is true if `SIMDUTF_IS_ZVBB` evaluates to true. Correct?

Reply:
Now, what you could do with `#if SIMDUTF_IS_ZVBB` that would be helpful to the code is to do...

```C++
#if SIMDUTF_IS_ZVBB
   bool supports_zvbb() const { return true; }
#else
  const bool _supports_zvbb;
   bool supports_zvbb() const { return _supports_zvbb; }
#endif
```

This would be better because if `SIMDUTF_IS_ZVBB` is true, then you'd avoid the runtime checks.

Reply:
Oh, yeah that's a better place to put the checks, I've adjusted the code.

Reply:
BTW you might think that the compiler would still optimize it away, but no... not even with -O3.
<img width="321" alt="Screenshot 2024-03-08 at 11 57 26 AM" src="https://github.com/simdutf/simdutf/assets/391987/44cbc901-abb1-46f9-a1a5-aa6e73b3b512">
<img width="326" alt="Screenshot 2024-03-08 at 11 57 22 AM" src="https://github.com/simdutf/simdutf/assets/391987/9a574a2d-02a2-48f0-bc46-81c29ff3c7c8">